presto-jmx/src/main/java/com/facebook/presto/connector/jmx/JmxMetadata.java

```diff
@@ -114,33 +121,61 @@ private JmxTableHandle getJmxHistoryTableHandle(SchemaTableName tableName)
     private JmxTableHandle getJmxTableHandle(SchemaTableName tableName)
     {
         try {
-            String canonicalName = new ObjectName(tableName.getTableName()).getCanonicalName();
-            Optional<ObjectName> objectName = mbeanServer.queryNames(WILDCARD, null).stream()
-                    .filter(name -> canonicalName.equalsIgnoreCase(name.getCanonicalName()))
-                    .findFirst();
-            if (!objectName.isPresent()) {
+            String objectNamePattern = toPattern(tableName.getTableName()); //toLowerCase(ENGLISH));
+            List<ObjectName> objectNames = mbeanServer.queryNames(WILDCARD, null).stream()
+                    .filter(name -> name.getCanonicalName().toLowerCase(ENGLISH).matches(objectNamePattern))
+                    .collect(toImmutableList());
+            if (objectNames.isEmpty()) {
                 return null;
             }
-            MBeanInfo mbeanInfo = mbeanServer.getMBeanInfo(objectName.get());
-
-            ImmutableList.Builder<JmxColumnHandle> columns = ImmutableList.builder();
+            List<JmxColumnHandle> columns = new ArrayList<>();
             columns.add(new JmxColumnHandle(NODE_COLUMN_NAME, createUnboundedVarcharType()));
+            columns.add(new JmxColumnHandle(OBJECT_NAME_NAME, createUnboundedVarcharType()));
+            for (ObjectName objectName : objectNames) {
+                MBeanInfo mbeanInfo = mbeanServer.getMBeanInfo(objectName);
+
+                getColumnHandles(mbeanInfo).forEach(columns::add);
+            }
 
             // Since this method is being called on all nodes in the cluster, we must ensure (by sorting)
             // that attributes are in the same order on all of them.
-            Arrays.stream(mbeanInfo.getAttributes())
-                    .filter(MBeanAttributeInfo::isReadable)
-                    .map(attribute -> new JmxColumnHandle(attribute.getName(), getColumnType(attribute)))
+            columns = columns.stream()
+                    .distinct()
                     .sorted(comparing(JmxColumnHandle::getColumnName))
-                    .forEach(columns::add);
+                    .collect(toImmutableList());
 
-            return new JmxTableHandle(tableName, ImmutableList.of(objectName.get().toString()), columns.build(), true);
+            return new JmxTableHandle(tableName, objectNames.stream().map(ObjectName::toString).collect(toImmutableList()), columns, true);
         }
         catch (JMException e) {
             return null;
         }
     }
 
+    private String toPattern(String tableName)
+            throws MalformedObjectNameException
+    {
+        if (!tableName.contains("*")) {
+            return new ObjectName(tableName).getCanonicalName();
+        }
+        String pattern = Pattern.compile("\\*").splitAsStream(tableName)
```

Comment:
I you used splitter.on('*') you would not need to do the extra checks for beginning and end of pattern below.